code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -222,21 +223,25 @@
 	..(gibbed)
 
 /mob/living/simple_animal/hostile/update_stamina()
-	. = ..()
-	move_to_delay = (initial(move_to_delay) + (staminaloss * 0.06))
-	UpdateSpeed()
+	if(staminaloss == 0)
+		remove_movespeed_modifier(/datum/movespeed_modifier/hostile_stamina_loss, TRUE)
+	else
+		add_or_update_variable_movespeed_modifier(/datum/movespeed_modifier/hostile_stamina_loss, TRUE, staminaloss * 0.06)
 
+///Use ChangeMoveToDelayBy instead for proper permanent speed changes
 /mob/living/simple_animal/hostile/proc/SpeedChange(amount = 0)
```

Comment:
So is this for altering the base speed without any modifiers or is this a proc that cant be reworked.

Reply:
After initialization move_to_delay will be calculated from all the movespeed modifiers, so using this would set the speed modifier to be the sum of all modifiers combined